spec/unit/util/pidlock_spec.rb

```diff
@@ -9,6 +9,39 @@
   before(:each) do
     @lockfile = tmpfile("lock")
     @lock = Puppet::Util::Pidlock.new(@lockfile)
+    @ps_argument_for_current_kernel = @lock.send(:ps_argument_for_current_kernel)
+  end
+
+  describe "#ps pid argument on posix" do
+    it "should be -eq when kernel is Linux" do
+      allow(Facter).to receive(:value).with(:kernel).and_return("Linux")
+      expect(@lock.send(:ps_argument_for_current_kernel)).to eq("-eq")
+    end
+
+    it "should be -T when kernel is AIX" do
+      allow(Facter).to receive(:value).with(:kernel).and_return("AIX")
+      expect(@lock.send(:ps_argument_for_current_kernel)).to eq("-T")
+    end
+
+    it "should be -p when kernel is Darwin or other" do
+      allow(Facter).to receive(:value).with(:kernel).and_return("Darwin")
+      expect(@lock.send(:ps_argument_for_current_kernel)).to eq("-p")
+    end
```

Comment:
There's a lot of this kind of thing in puppet, but I try to avoid testing private methods, since it's testing the implementation instead of the behavior, as it makes it harder to change in the future and it's usually a double bookkeeping kind of test.

I'd drop these and in the "with a stale lock" example group, test how the "should replace with new locks" test behaves on Linux, AIX, Darwin, Windows.

Reply:
The behaviour remains pretty much the same (at least the initially expected one). The implementation is the one that changed actually. 
Agreed on double bookkeeping so I've replaced those tests. One ensures the ps parameter used on current kernel is a valid one and the other is for the fallback mechanism created for Linux operating systems which have outdated ps.
I don't think other unit tests will bring any added value. An acceptance test maybe could.